step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AbstractAutomationPackageReader.java

```diff
@@ -146,51 +149,63 @@ private void fillAutomationPackageWithAnnotatedKeywordsAndPlans(AutomationPackag
+                    if (!isLocalPackage) {
+                        String functionName = annotation.name().length() > 0 ? annotation.name() : m.getName();
 
-            Function f;
-            if (isCompositeFunction(annotation)) {
-                f = createCompositeFunction(m, annotation);
-            } else {
-                if (!isLocalPackage) {
-                    String functionName = annotation.name().length() > 0 ? annotation.name() : m.getName();
+                        GeneralScriptFunction function = new GeneralScriptFunction();
+                        function.setAttributes(new HashMap<>());
+                        function.getAttributes().put(AbstractOrganizableObject.NAME, functionName);
 
-                    GeneralScriptFunction function = new GeneralScriptFunction();
-                    function.setAttributes(new HashMap<>());
-                    function.getAttributes().put(AbstractOrganizableObject.NAME, functionName);
+                        // to be filled by AutomationPackageKeywordsAttributesApplier
+                        if (scriptFile != null) {
+                            function.setScriptFile(new DynamicValue<>(scriptFile));
+                        }
 
-                    // to be filled by AutomationPackageKeywordsAttributesApplier
-                    if (scriptFile != null) {
-                        function.setScriptFile(new DynamicValue<>(scriptFile));
-                    }
+                        // libraries file is not used is automation package (only required for compatibility with StepJarParser)
+                        if (librariesFile != null) {
+                            function.setLibrariesFile(new DynamicValue<>(librariesFile));
+                        }
 
-                    // libraries file is not used is automation package (only required for compatibility with StepJarParser)
-                    if (librariesFile != null) {
-                        function.setLibrariesFile(new DynamicValue<>(librariesFile));
+                        function.getCallTimeout().setValue(annotation.timeout());
+                        function.setScriptLanguage(new DynamicValue<>("java"));
+                        f = function;
+                    } else {
+                        f = LocalFunctionPlugin.createLocalFunction(m, annotation);
                     }
+                }
 
-                    function.getCallTimeout().setValue(annotation.timeout());
-                    function.setDescription(annotation.description());
-                    function.setScriptLanguage(new DynamicValue<>("java"));
-                    f = function;
-                } else {
-                    f = LocalFunctionPlugin.createLocalFunction(m, annotation);
+                f.setDescription(annotation.description());
+                f.setSchema(annotatedKeywordJsonSchemaCreator.createJsonSchemaForKeyword(m));
+
+                String htmlTemplate = f.getAttributes().remove("htmlTemplate");
```

Comment:
This also still exists in KW package but we should completely remove htmlTemplate at some point (the functionality is not supported anymore).